### PR TITLE
feat(frontend): use Backend API via canister

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

The idea is to create a Backend canister class (instead of using actors) and refactor `backend.api` methods to use it.

### CODE REVIEW

File: src/frontend/src/eth/services/erc20-user-tokens-services.ts [hunk 1]
```diff
@@ -26,7 +28,8 @@ export const saveUserTokens = async ({
 
 	await setManyUserTokens({
 		identity,
-		tokens: tokens.map(toUserToken)
+		tokens: tokens.map(toUserToken),
+		nullishIdentityErrorMessage: get(i18n).auth.error.no_internet_identity
```
[the remark this answers is not in the document]
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/eth/services/erc20-user-tokens-services.ts:32]: Sounds good! Will do.

File: src/frontend/src/lib/api/backend.api.ts [hunk 2]
```diff
@@ -6,109 +6,122 @@ import type {
...
-import type { Identity } from '@dfinity/agent';
-import type { Principal } from '@dfinity/principal';
-import { toNullable, type QueryParams } from '@dfinity/utils';
+import { BackendCanister } from '$lib/canisters/backend.canister';
+import { BACKEND_CANISTER_ID } from '$lib/constants/app.constants';
+import type { CommonCanisterApiFunctionParams } from '$lib/types/canister';
+import { Principal } from '@dfinity/principal';
+import { assertNonNullish, isNullish, type QueryParams } from '@dfinity/utils';
+
+export type AddUserCredentialParams = {
```
[the remark this answers is not in the document]
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/api/backend.api.ts]: Yes, this one and the ones below are used in a few places (canister, api, tests). Moved them to `types/api`.

File: src/frontend/src/lib/api/backend.api.ts [hunk 3]
```diff
@@ -6,109 +6,122 @@ import type {
...
+
+let canister: BackendCanister | undefined = undefined;
 
 export const listUserTokens = async ({
 	identity,
-	certified = true
-}: { identity: OptionIdentity } & QueryParams): Promise<UserToken[]> => {
-	const { list_user_tokens } = await getBackendActor({ identity, certified });
-	return list_user_tokens();
+	certified
```
[the remark this answers is not in the document]
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/api/backend.api.ts]: Absolutely, moved it back for now.

File: src/frontend/src/lib/api/signer.api.ts [hunk 4]
```diff
@@ -2,31 +2,26 @@ import type { BitcoinNetwork, SignRequest } from '$declarations/signer/signer.di
...
-	identity: OptionIdentity;
-	nullishIdentityErrorMessage?: string;
-	signerCanisterId?: CanisterIdText;
-};
-
 export const getBtcAddress = async ({
 	identity,
 	network
-}: CommonParams<{
+}: CommonCanisterApiFunctionParams<{
```
[the remark this answers is not in the document]
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/api/signer.api.ts]: Agreed, all changes to the signer have been removed from this PR.

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

**Pull request merged**